### PR TITLE
[SRK-11] Run HLint during Travis build, and fail if it finds any hints

### PR DESCRIPTION
**Pull request by neongreen**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -31,7 +31,12 @@ before_install:
 # TODO: https://github.com/NixOS/nixpkgs/issues/21200
 - travis_retry curl -L https://www.stackage.org/stack/$TRAVIS_OS_NAME-x86_64 | tar
   xz --strip-components=1 -C ~/.local/bin
+
 script:
+- pushd ~/.local    # any path that doesn't have a stack.yaml
```
Comment (domenkozar) [inline .travis.yml]: Why is this needed, can't we just install into the project itself?

**Comment by domenkozar** [comment]
@neongreen resolve conflicts first :) otherwise pretty good!

File: .travis.yml (hunk 1 above)
Comment (neongreen) [inline .travis.yml]: No, because we override the `haskell-src-exts` version in the project and hlint doesn't work with the older version we're using. I'll add a comment to `.travis.yml` about that.

Comment (domenkozar) [inline .travis.yml]: We don't need to override it anymore as `derive-2.6.2` was released

Comment (neongreen) [inline .travis.yml]: Done.

**Comment by neongreen** [comment]
Resolved conflicts, waiting for Travis.

**Comment by domenkozar** [comment]
@neongreen so can we use newer derive and remove `haskell-src-exts`?

**Comment by neongreen** [comment]
@domenkozar maybe. I'll look into it.

**Comment by neongreen** [comment]
@domenkozar okay, I did that.
